### PR TITLE
Adds new Sign up form 

### PR DESCRIPTION
**Pull request by charlottebrf**
Working on issue #18. Used existing default and gallery html files as template guides. Attempted to add in a sign_up_form.html to layouts folder &  sign_up_form.html to img folder. 

* [x] This feature remains incomplete, need to work out why the new sign up page created isn’t appearing. @eddiejaoude advice on Jekyll would be great!

# Tasks to be done:

1. [x] Create type form for sign up -created a google form: https://docs.google.com/forms/d/e/1FAIpQLSftbQrHueHZpjbXyXhWKvR-uixs2J_BFhxJL29ZVp-v9vFoiQ/viewform?embedded=true
2. [x] Create page for sign up form -see screen shot this now appears correctly
3. [x] Linked from main navigation - see screen shot this is now linked
3. [x] Embed form on new page  - replaced the google form with a [Formspree form](https://formspree.io). Users can now submit contact details which are then emailed to a Hack Brexit gmail account & then need to be stored by an administrator. 
4. [ ] Explore alternatives for sign up form so that data could be stored in a JSON file or a database so that data collection is more efficient, and doesn't depend on an Admin checking & collating emails from Formspree. 

hack brexit sign up form created

### CODE REVIEW

**Comment by eddiejaoude** [comment]
@charlottebrf  To add a button/link to the navigation you will need to update `_includes/header.html`

**Comment by eddiejaoude** [comment]
@charlottebrf also the `sign_up_form.html` in the root of the project includes the wrong template, currently it is `default`...

```
---
layout: sign_up_form
---
```

File: sign_up_form.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+layout: Sign up form
```
Comment (eddiejaoude) [inline sign_up_form.html]: This will need to be the name of the `html` layout file (without the `.html`). In this case `sign_up_form`.

Comment (charlottebrf) [reply sign_up_form.html]: Thanks @eddiejaoude that's been changed

**Comment by charlottebrf** [comment]
@eddiejaoude Am I ok to merge this PR?

**Comment by eddiejaoude** [comment]
@charlottebrf not yet. Needs to be reviewed first 🤓 . 

Questions:
- It has the label `incomplete`, is it now complete & ready for review? If so, please review the `incomplete` label
- In the description there is 1 task unchecked - is it still pending or will it be split out to another issue / pull request?

**Comment by eddiejaoude** [comment]
I tried signing up, it doesn't seem to send it to my email that I signed up with, but the hackbrexit email? 

hackbrexit

**Comment by charlottebrf** [comment]
Hey @eddiejaoude yes that's correct. I used Formspree to collect the email addresses so an email is sent to the Hack Brexit email address, but the user doesn't get an automatic email. Instead at this level it requires an admin to check the emails & manually update a spreadsheet then email everyone. 

Not 100% ideal. The unchecked tick box would be to make it a better process.In the meantime if you think it's good enough, we can publish this & then I create a separate issue to improve the process. Removed the Incomplete label, but I'd left that as was unsure initially it's considered completed.

**Comment by eddiejaoude** [comment]
@charlottebrf ok, just checking 👍 

The only other suggestion I would make is maybe use some bootstrap styling as bootstrap is already available on the page. Here is a link to the [bootstrap v4 form docs](http://v4-alpha.getbootstrap.com/components/forms/)

**Comment by charlottebrf** [comment]
@eddiejaoude When I look at the _config.yml file I can't see Bootstrap added. On Stackoverflow it advises adding Bootstrap to this file: http://stackoverflow.com/questions/28733425/adding-bootstrap-to-jekyll

Should I do that before trying to add bootstrap to the form? I can see Bootstrap in the head.html file though. Is this what you were referring to? 
bootstrap


screen shot 2016-11-26 at 10 12 06

**Comment by charlottebrf** [comment]
screen shot 2016-11-26 at 10 43 56

 @eddiejaoude Bootstrap added & form is working. Look ok to you?

**Comment by eddiejaoude** [comment]
> Should I do that before trying to add bootstrap to the form? I can see Bootstrap in the head.html file though. Is this what you were referring to?

Yep, it is already include 😄

**Comment by eddiejaoude** [comment]
Looks great 👍

File: _layouts/sign_up_form.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<div class="container">
+    {% include header.html %}
+
+<h1>Hack Brexit Mailing List</h1>
+
+<p>Sign up on our Hack Brexit contact form for us to keep you updated with all of the latest news </p>
+
+<form method="POST" action="http://formspree.io/hackbrexit@gmail.com">
+
+<form class="form-inline">
```
Comment (eddiejaoude) [inline _layouts/sign_up_form.html]: Nested form tags is not correct. This is the 2nd one, you can combine them both.

**Comment by charlottebrf** [comment]
screen shot 2016-11-26 at 11 00 51


Having nested this has now changed the view of the form. @eddiejaoude did I do something wrong?

**Comment by eddiejaoude** [comment]
Its because you are using `form-inline`

**Comment by eddiejaoude** [comment]
I made the changes in another PR that is requesting to be merged into you PR, so it didnt cause you conflicts & you could easily see the changes & you could choose to accept or not.

Hope this makes it more clear...

* Your PR: `features/18` -> `master`
* My PR: `features/18-tweaks` -> `feature/18` - @charlottebrf you can close or accept this one as it is not going to `master` but to your branch https://github.com/HackBrexit/hackbrexit.github.io/pull/30

**Comment by charlottebrf** [comment]
Thanks @eddiejaoude now this is pushing to feature/18 are we now set to merge this PR?

**Comment by eddiejaoude** [comment]
Looks good @charlottebrf 👍

**Comment by eddiejaoude** [comment]
@charlottebrf Give it a few minutes, than can you re-test on live?

**Comment by charlottebrf** [comment]
@eddiejaoude Great thanks! Yes re-tested & appears to be working fine. Shall I now close this PR?

**Comment by eddiejaoude** [comment]
I already merged the PR, then it got deployed, so thats it for this PR 😄

**Pull request merged**